fixtures/ios_app_with_templates/Tuist/Templates/custom/Template.swift

```diff
@@ -0,0 +1,20 @@
+import ProjectDescription
+
+let nameAttribute: Template.Attribute = .required("name")
+let platformAttribute: Template.Attribute = .optional("platform", default: "ios")
+
+let testContents = """
+// this is test \(nameAttribute) content
+"""
+
+let template = Template(
+    description: "Custom template",
+    attributes: [
+        nameAttribute,
+        platformAttribute
+    ],
+    files: [
+        .string(path: "custom_dir/custom.swift", contents: testContents)
+    ],
+    directories: ["custom_dir"]
```

Comment:
I'd get rid of the `directories` attribute from both the model in `ProjectDescriptionHelpers` and `TuistCore`. As a user, if I indicate that I'd like the file to be created under `custom_dir/custom.swift` I'd expect `custom_dir` to be created automatically. I don't think there's a case where the users will want to create an empty directory. If it turns out that we are wrong in this assumption, we can add support for that in the future.